stix_shifter_modules/guardium/stix_transmission/api_client.py

```diff
@@ -1,250 +1,73 @@
-        password = configuration.get('auth').get("password", None)
-        grant_type = connection.get('options',{}).get("grant_type", 'password')
-        client_id = connection.get('options',{}).get("client_id", None)
-        client_secret = connection.get('options',{}).get("client_secret", None)
-        # It is decided the authorization will not be sent by UDS
-
-        if(username is None or password is None or grant_type is None or client_id is None or client_secret is None):
-            self.credential = None
-            raise IOError(
-                    3001, "Guardium Credential not provided in the connection / configuration object")
-        else:
-            self.credential = {"username": username, "password": password, "grant_type": grant_type,
-                               "client_id": client_id, "client_secret": client_secret}
-
-        host_port = connection.get('host') + ':' + \
-            str(connection.get('port', ''))
-
-        url_modifier_function = None
-        proxy = connection.get('proxy')
-        if proxy is not None:
-            proxy_url = proxy.get('url')
-            proxy_auth = proxy.get('auth')
-            if (proxy_url is not None and proxy_auth is not None):
-                self.headers['proxy'] = proxy_url
-                self.headers['proxy-authorization'] = 'Basic ' + proxy_auth
-            if proxy.get('x_forward_proxy', None) is not None:
-                self.headers['x-forward-url'] = 'https://' + \
-                    host_port + '/'  # + endpoint, is set by 'add_endpoint_to_url_header'
-                host_port = proxy.get('x_forward_proxy')
-                if proxy.get('x_forward_proxy_auth', None) is not None:
-                    self.headers['x-forward-auth'] = proxy.get(
-                        'x_forward_proxy_auth')
-                self.headers['user-agent'] = 'UDS'
-                url_modifier_function = self.add_endpoint_to_url_header
-
-        self.client = RestApiClient(host_port,
-                                    None,
-                                    self.headers,
-                                    url_modifier_function,
-                                    cert_verify=connection.get(
-                                        'selfSignedCert', True),
-                                    sni=connection.get('sni', None)
-                                    )
-        self.search_timeout = connection['options'].get('timeout')
-
-    def add_endpoint_to_url_header(self, url, endpoint, headers):
-        # this function is called from 'call_api' with proxy forwarding,
-        # it concatenates the endpoint to the header containing the url.
-        headers['x-forward-url'] += endpoint
-        # url is returned since it points to the proxy for initial call
-        return url
-
-    def add_header_elements(self, key, value):
-        self.headers[key] = value
-        return
-
-    def ping_box(self):
```

Comment:
not sure why you remove this function. you need to make a call to the api in order to know if the api is accessible.

Reply:
You can see in entry_point.py  ping uses a different function

Reply:
yes I can see that but the implementation of that function is wrong. I mentioned that in another comment on that function.